core/Config.php

```diff
@@ -514,7 +514,7 @@ protected function getUncached($class, $name, $sourceOptions, &$result, $suppres
 		// Include extensions only if not flagged not to, and some have been set
 		if (($sourceOptions & self::EXCLUDE_EXTRA_SOURCES) != self::EXCLUDE_EXTRA_SOURCES) {
 			// If we don't have a fresh list of extra sources, get it from the class itself
-			if (!array_key_exists($class, $this->extraConfigSources)) {
+			if (!isset($class[$this->extraConfigSources]) || $class[$this->extraConfigSources] === null) {
```

Comment:
Tried writing an optimizer. Due to the is_array call count being trivial (ie. 1000 calls on vanilla Silverstripe), it didn't really do much and it's probably not worth. Not even going to bother with the rest as they're all also under 1000 calls.